app/src/main/java/com/android/sample/ui/authentication/SignIn.kt

```diff
@@ -0,0 +1,226 @@
+              modifier = Modifier.fillMaxWidth(0.8f).testTag("PasswordTextField"))
+          // Display password error message below the password field
+          if (passwordErrorState.value != null) {
+            Text(
+                text = passwordErrorState.value ?: "",
+                color = Color.Red,
+                fontSize = 12.sp,
+                modifier = Modifier.align(Alignment.Start).padding(start = 40.dp, top = 4.dp))
+          }
+          Spacer(modifier = Modifier.height(16.dp))
+
+          // Sign In with Email/Password Button
+          Button(
+              onClick = {
+                when {
+                  passwordState.value.isEmpty() -> {
+                    passwordErrorState.value =
+                        "Password cannot be empty" // Set the error message if password is empty
+                  }
+                  else -> {
+                    signInWithEmailAndPassword(emailState.value, passwordState.value, context)
+                  }
+                }
+              },
+              modifier = Modifier.fillMaxWidth(0.8f).height(48.dp).testTag("SignInButton")) {
+                Text("Sign in with Email", fontSize = 16.sp)
+              }
+          Spacer(modifier = Modifier.height(16.dp))
+
+          // Authenticate With Google Button
+          GoogleSignInButton(
+              onSignInClick = {
+                val gso =
+                    GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                        .requestIdToken(token)
+                        .requestEmail()
+                        .build()
+                val googleSignInClient = GoogleSignIn.getClient(context, gso)
+                launcher.launch(googleSignInClient.signInIntent)
+              })
+
+          // If user already has an account, navigate to the sign in screen
+          TextButton(
+              onClick = { navigationActions.navigateTo(Screen.SIGN_UP) },
+              modifier = Modifier.fillMaxWidth(0.8f).height(36.dp).testTag("GoToSignUpButton")) {
+                Text("No account yet?", fontSize = 16.sp)
+              }
+
+          // Continue as a guest button
+          TextButton(
+              onClick = { navigationActions.navigateTo(Screen.OVERVIEW) },
+              modifier =
+                  Modifier.fillMaxWidth(0.8f).height(36.dp).testTag("ContinueAsGuestButton")) {
+                Text("Continue as a guest", fontSize = 16.sp)
+              }
+        }
+      })
+}
+
+fun signInWithEmailAndPassword(email: String, password: String, context: Context) {
+  val auth = Firebase.auth
+  auth.signInWithEmailAndPassword(email, password).addOnCompleteListener { task ->
+    if (task.isSuccessful) {
+      Log.d("SignInScreen", "signInWithEmail:success")
+      Toast.makeText(context, "Email login successful!", Toast.LENGTH_LONG).show()
+      // Navigate to the next screen upon successful login
```

Comment:
Don't forget to navigate to the screen, now that you have the navigationController